Sources/Penny/Services/CacheService/CacheStorage.swift

```diff
@@ -0,0 +1,30 @@
+import Models
+
+struct CacheStorage: Codable {
+
+    struct ProposalsCheckerStorage: Codable {
+        var previousProposals: [Proposal]
+        var queuedProposals: [QueuedProposal]
+    }
+
+    var reactionCacheData: ReactionCache.Storage?
+    var proposalsCheckerData: ProposalsCheckerStorage?
+
+    init() { }
+
+    static func makeFromCachedData() async -> CacheStorage {
+        var storage = CacheStorage()
+        storage.reactionCacheData = await ReactionCache.shared.toCacheStorageData()
+        storage.proposalsCheckerData = await ProposalsChecker.shared.toCacheStorageData()
+        return storage
+    }
+
+    func populateServices() async {
+        if let reactionCacheData = self.reactionCacheData {
+            await ReactionCache.shared.fromCacheStorageData(reactionCacheData)
```

Comment:
`await whatever.fromWhatever()` is not a particularly idiomatic formation. "From the cache storage data" _to where_? What is the verb, what action is being taken? A method with a name like this would typically be a factory method and/or return a value.